lib/src/platform_check/web.dart

```diff
@@ -16,7 +16,8 @@ class PlatformWeb extends Platform {
     try {
       Random.secure();
       useBuiltInRng = true;
-    } on UnsupportedError {
+    } catch (_) {
+      // throws UnknownJsTypeError. See 'dart:_js_types'.
       useBuiltInRng = false;
     }
```

Comment:
Suggestion: Catch the specific known errors so as not to squash unknown exceptions accidentally.
```suggestion
    } on UnsupportedError {
      useBuiltInRng = false;
    } on UnknownJsTypeError {
      // Short explanation on why this case exists and why it is ok to set false in that case.
      useBuiltInRng = false;
    }
```

Is it possible to make this more specific?

Reply:
`UnknownJsTypeError` comes from the  ~dart:_js_types~ `dart:_js_helper` library which is internal and cannot be imported.

Reply:
Aah, then this effort might be futile. Or is there a way we could uniquely identify it? By some form of message?

Reply:
we can use something like this

```dart
} catch (e) {
  if (e.runtimeType.toString() == 'UnknownJsTypeError') {
  // most likely cause 
  }
  else { 
  // in case Dart changes it to some other error type in the future.
  }
}
```

Reply:
That is probably better.

But I would keep the first `on UnsupportedError` and then throw in the else statement to fail fast.